geth/rpc/route.go

```diff
@@ -15,60 +15,73 @@ func newRouter(upstreamEnabled bool) *router {
 		upstreamEnabled: upstreamEnabled,
 	}
 
-	for _, m := range localMethods {
+	for _, m := range remoteMethods {
 		r.methods[m] = true
 	}
 
 	return r
 }
 
-// routeLocally returns true if given method should be routed to
-// the local node
-func (r *router) routeLocally(method string) bool {
-	// if upstream is disabled, always route to
-	// the local node
+// routeRemote returns true if given method should be routed to the remote node
+func (r *router) routeRemote(method string) bool {
 	if !r.upstreamEnabled {
-		return true
+		return false
 	}
 
 	// else check route using the methods list
 	return r.methods[method]
 }
 
-// localMethods contains methods that should be routed to
-// the local node; the rest is considered to be routed to
-// the upstream node.
-// TODO: in the future, default option may be "local node",
-// so it'd be convinient to keep track of "remoteMethods" here.
-var localMethods = [...]string{
-	//Whisper commands
-	"shh_post",
-	"shh_version",
-	"shh_newIdentity",
-	"shh_hasIdentity",
-	"shh_newGroup",
-	"shh_addToGroup",
-	"shh_newFilter",
-	"shh_uninstallFilter",
-	"shh_getFilterChanges",
-	"shh_getMessages",
-
-	// DB commands
-	"db_putString",
-	"db_getString",
-	"db_putHex",
-	"db_getHex",
-
-	// Other commands
-	"net_version",
-	"net_peerCount",
-	"net_listening",
-
-	// blockchain commands
-	"eth_sign",
-	"eth_accounts",
-	"eth_getCompilers",
-	"eth_compileLLL",
-	"eth_compileSolidity",
-	"eth_compileSerpent",
+// remoteMethods contains methods that should be routed to
+// the upstream node; the rest is considered to be routed to
+// the local node.
+// TODO(tiabc): Write a test on each of these methods to ensure they're all routed to the proper node and ensure they really work.
```

Comment:
Isn't `TestRouteWithUpstream` and `TestRouteWithoutUpstream` already doing that?

Reply:
Yes but not exactly. I want take every method and run it. This thing is sort of serious.